ProcessLib/Process.h

```diff
@@ -200,6 +203,10 @@ class Process
 
     VectorMatrixAssembler _global_assembler;
 
+    /// Coupled processes.
+    mutable std::weak_ptr<StaggeredCouplingTerm> _coupling_term;
```

Comment:
It is a common question.
Such case is included in struct StaggeredCouplingTerm, which has only two members to store the references to the coupled processes and the references to the solutions of the coupled processes:
```
    /// References to the coupled processes are distinguished by the keys of
    /// process types.
    std::unordered_map<std::type_index, Process const&> const&
        coupled_processes;

    /// References to the current solutions of the coupled processes.
    /// The coupled solutions are distinguished by the keys of process types.
    std::unordered_map<std::type_index, GlobalVector const&> const& coupled_xs;
```